src/main/java/org/apache/sling/pipes/internal/MovePipe.java

```diff
@@ -62,24 +67,29 @@ public boolean modifiesContent() {
             String targetPath = getExpr();
             try {
                 Session session = resolver.adaptTo(Session.class);
-                if (session.itemExists(targetPath)){
+                if (orderBefore) {
+                    output = reorder(resource, targetPath, session);
+                } else if (session.itemExists(targetPath)) {
                     if (overwriteTarget && !isDryRun()) {
```

Comment:
i know it's not you, but could you move isDryRun test to after the log is emitted?